### ISSUE TITLE
Ideal-image: support non-blurry images in dev

### ISSUE THREAD

**Issue by Irev-Dev** [facebook/docusaurus#4588]
## 🐛 Bug Report

`@docusaurus/plugin-ideal-image` doesn't work properly in dev, it never loads the full resolution images leaving only the blurry image like so.
![image](https://user-images.githubusercontent.com/29681384/114086811-73f8e400-98f6-11eb-82d5-793d90aaa018.png)

It appears to be fine for production/when built.

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## To Reproduce

```bash
npx @docusaurus/init@latest init my-website classic
yarn add @docusaurus/plugin-ideal-image
````
add `plugins: ['@docusaurus/plugin-ideal-image'],` to `docusaurus.config.js`

add the following somewhere in `markdown-features.mdx`
```jsx
import Image from '@theme/IdealImage';
import thumbnail from '../static/img/docusaurus.png';
<Image img={thumbnail} />
```

`yarn start`

navigate to http://localhost:3000/docs/markdown-features

## Expected behavior

full resolution image would load

## Actual Behavior

it remains blurry (and adds a bunch of white space below the image that's not there when the project is built)
![image](https://user-images.githubusercontent.com/29681384/114087561-609a4880-98f7-11eb-9540-9aa258211692.png)

The docusaurus logo is easily identifiable, but in a real scenario it's impossible to determine what the images are and so it so I end up building the project to review the docs. Here's an example of what that looks like.
![image](https://user-images.githubusercontent.com/29681384/116390760-b8c2cb80-a861-11eb-9409-8fe361984034.png)

## Your Environment

- Docusaurus version used: 2.0.0-alpha.72
- Environment name and version: chrome Version 89.0.4389.114 (Official Build) (x86_64), Node v14.16.0
- Operating system and version (desktop or mobile): Mac Big Sur 11.1

## Reproducible Demo

**not possible it seems** on codersandbox, I'm not sure if they do some dependencies magic that breaks with ideal image.
Here's a link anyway https://codesandbox.io/s/admiring-bell-phfxj?file=/docs/markdown-features.mdx
but it has the following error `Module not found: Can't resolve '@theme/IdealImage' in '/sandbox/docs'`

This seems like a different issue to #4255

**Comment by slorber** [comment]
Not 100% sure because the original author is [not there anymore](https://docusaurus.io/blog/2020/01/07/tribute-to-endi) but I guess it was on purpose, as we don't want the dev server to resize all the images and slow down the DX.

If anyone wants to add support non-blurry ideal images in dev, PRs are welcome as long as they don't degrade the DX.

**Comment by Irev-Dev** [comment]
Okay thanks @slorber 

I'll add that when ever you navigate to a page with an ideal image on it there is a cryptic error in the console too.
```:3000/docs/:1 Uncaught (in promise)```

**Comment by lukaskoeller** [comment]
I am experiencing the same behavior in development. Image stays blurry and there is an error saying `Uncaught (in promise)`.
<img width="639" alt="Screenshot 2021-04-11 at 19 55 25" src="https://user-images.githubusercontent.com/10390627/114315568-de30a500-9aff-11eb-9806-cb25320ae22e.png">

**Comment by martis-git** [comment]
Still actual!

![image](https://user-images.githubusercontent.com/42924400/124378397-c6f5f380-dcb9-11eb-88a0-423b463fc5fe.png)

**Comment by Irev-Dev** [comment]
These days I'm just running `yarn build && yarn serve` for dev and rebuild each time I want to see updates as I find `yarn dev` unusable since [my site](https://learn.cadhub.xyz/blog/right-level-of-abstraction) has a lot of images.

**Comment by lex111** [comment]
This issue is really annoying, so I suggest not resizing images during development, will be fixed in #5540.

**Comment by Irev-Dev** [comment]
Thanks @lex111 😍
